test/module/irohad/ordering/on_demand_os_test.cpp

```diff
@@ -197,40 +197,54 @@ TEST_F(OnDemandOsTest, DISABLED_ConcurrentInsert) {
 /**
  * @given initialized on-demand OS
  * @when  insert proposal_limit rounds twice
- * @then  on second rounds check that old proposals are expired
+ * @then  old proposals are expired
  */
 TEST_F(OnDemandOsTest, Erase) {
   for (auto i = commit_round.block_round;
        i < commit_round.block_round + proposal_limit;
        ++i) {
     generateTransactionsAndInsert({i + 1, commit_round.reject_round}, {1, 2});
     os->onCollaborationOutcome({i, commit_round.reject_round});
-    ASSERT_TRUE(os->onRequestProposal({i + 1, commit_round.reject_round}));
   }
 
   for (consensus::BlockRoundType i = commit_round.block_round + proposal_limit;
        i < commit_round.block_round + 2 * proposal_limit;
        ++i) {
     generateTransactionsAndInsert({i + 1, commit_round.reject_round}, {1, 2});
     os->onCollaborationOutcome({i, commit_round.reject_round});
-    ASSERT_FALSE(os->onRequestProposal(
-        {i + 1 - proposal_limit, commit_round.reject_round}));
+  }
+
+  consensus::Round last_round = {commit_round.block_round + 2 * proposal_limit,
+                                 commit_round.reject_round};
+  for (consensus::BlockRoundType i = commit_round.block_round;
+       i < commit_round.block_round + 2 * proposal_limit;
+       ++i) {
+    consensus::Round current_round = {i + 1, commit_round.reject_round};
+    auto proposal = os->onRequestProposal({i + 1, commit_round.reject_round});
+
+    if (current_round.block_round >= last_round.block_round) {
```

Comment:
Seems, if else ladder could be refactored with inline function and used twice for this and `EraseReject` tests.

Reply:
Well, it's possible but these 2 code fragments are actually a bit different (using `.block_round` in the first one instead of `.reject_round` in the second) so I guess the resulting code will be too difficult to read.